packages/react-lab/src/date-picker/Calendar/Calendar.js

```diff
@@ -0,0 +1,104 @@
+import { Box } from '@tonic-ui/react';
+import { usePrevious } from '@tonic-ui/react-hooks';
+import isSameMonth from 'date-fns/isSameMonth';
+import isSameYear from 'date-fns/isSameYear';
+import React, { forwardRef, useEffect, useState } from 'react';
+import DecadeView from './DecadeView';
+import MonthView from './MonthView';
+import YearView from './YearView';
+import Navigation from './Navigation';
+import { useCalendarStyle } from './styles';
+import {
+  isDate,
+  toDate,
+} from '../utils';
+
+const Calendar = forwardRef((
+  {
+    calendarStartDay = 0, // 0 = Sunday, 1 = Monday, ...
+    children,
+    dateFormat = 'yyyy-MM-dd',
+    value = new Date(),
+    view = 'month', // one of 'month', 'year', 'decade'
+    onChange: onClickDay,
+    ...props
+  },
+  ref,
+) => {
+  const currentDate = isDate(value) ? toDate(value) : new Date();
+  const inputDate = isDate(value) ? toDate(value) : null;
+  const [currentView, setView] = useState(view);
+  const [activeDate, setActiveDate] = useState(currentDate);
+  const [calendarValue, setCalendarValue] = useState(inputDate);
+  const styleProps = useCalendarStyle();
+  const previouslyValue = usePrevious(value);
+
+  useEffect(() => {
+    // Dynamically change the calendar view
+    const isValueChange = !!inputDate && value !== previouslyValue;
+    const needToChangeView = !isSameYear(inputDate, activeDate) || !isSameMonth(inputDate, activeDate);
+    if (isValueChange && needToChangeView) {
+      setActiveDate(inputDate);
+    }
+    if (isValueChange) {
+      setCalendarValue(inputDate);
+    }
+  }, [value, previouslyValue, inputDate, activeDate, setActiveDate, setCalendarValue]);
+
+  if (children) {
+    return (
+      <Box
+        ref={ref}
+        {...styleProps}
+        {...props}
+      >
+        {children}
+      </Box>
+    );
+  }
+
+  return (
+    <Box
+      ref={ref}
+      {...styleProps}
+      {...props}
+    >
+      <Navigation
```

Comment:
How about moving these getter/setter props to a context rather than directly passing them to its descendant child components?

Reply:
For example:
![image](https://user-images.githubusercontent.com/447801/159670972-15c2c179-5b2f-40cf-8c91-e5926829544a.png)